pkg/vsphere/disk/disk.go

```diff
@@ -45,6 +45,43 @@ type Filesystem interface {
 	Unmount(path string) error
 }
 
+// ReferenceCount represents the number of references to a disk
+type ReferenceCount struct {
```

Comment:
Ehh... just `Semaphore`?  Feels weird, but I'll go with it for now unless someone has something more fitting. I agree, though, this is a count of references, not a reference.